include/triton/Dialect/Triton/IR/TritonOps.td

```diff
@@ -882,9 +882,18 @@ def TT_GatherOp : TT_Op<"gather", [Pure,
     dimension, and each dimension of the indices tensor that is not the gather
     dimension cannot be greater than the corresponding dimension in the input
     tensor.
+
+    The `efficient_layout` attribute is set when the compiler has determined an
+    optimized layout for the operation, indicating that it should not be
+    changed.
   }];
 
-  let arguments = (ins TT_Tensor:$src, TT_IntTensor:$indices, I32Attr:$axis);
+  let arguments = (ins
+    TT_Tensor:$src,
+    TT_IntTensor:$indices,
+    I32Attr:$axis,
+    UnitAttr:$efficient_layout
```

Comment:
That's probably a good pattern! What else does `ttg.layout_anchor` do?

Reply:
I think they are different things

Reply:
By reading the code, I found that it's `getEfficientLayout` is also used to determine if `gather(cvt(src), idx) -> gather(src, idx)` can happen. So `layout_anchor` is more generic and probably doesn't fit his use case

Reply:
Yes but he doesn't want that pattern to happen because layout anchors shouldn't be folded into convert ops.

Reply:
@peterbell10 let me know when you landed the introduction of `ttg.layout_anchor` and I'll refactor this (ReshapeOp can also use it FWIW) I think this is the right pattern: when the compiler (or the user!) decides it can assign an optimized layout, we shouldn't change it.